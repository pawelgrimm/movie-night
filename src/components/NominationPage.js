import React, { useState } from "react";
import { connect } from "react-redux";
import MovieSearch from "./MovieSearch";
import MovieList from "./MovieList";
import VideoModal from "./VideoModal";
import ConfirmationModal from "./ConfirmationModal";
import { setMovies } from "../actions/movie";

const mapStateToProps = ({ movies }) => ({
  movies: movies,
});

const mapDispatchToProps = (dispatch) => ({
  clearMovieList: () => dispatch(setMovies([])),
});

export const NominationPage = ({ appElement, clearMovieList, movies = [] }) => {
  const [isConfirmationModalOpen, setConfirmationModalOpen] = useState(false);

  return (
    <div className="content-container">
      <MovieSearch numResults={5} />
      <MovieList />
      <div className="container--flex-row">
        <button className="button">Next (Voting)</button>
        <button
          className="button button--secondary"
          onClick={() => setConfirmationModalOpen(true)}
          disabled={movies.length < 1}
        >
          Start Over
        </button>
      </div>
      <VideoModal appElement={appElement} />
      <ConfirmationModal
        appElement={appElement}
        isOpen={isConfirmationModalOpen}
        onClose={() => setConfirmationModalOpen(false)}
        onConfirm={() => {
          clearMovieList();
          setConfirmationModalOpen(false);
        }}
      />
    </div>
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(NominationPage);
