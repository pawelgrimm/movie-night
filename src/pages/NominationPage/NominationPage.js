import React, { useEffect, useState } from "react";
import { connect } from "react-redux";
import MovieSearch from "../../containers/MovieSearch/MovieSearch";
import MovieList from "./components/NominationMovieList";
import VideoModal from "../../containers/VideoModal/VideoModal";
import ConfirmationModal from "../../components/ConfirmationModal/ConfirmationModal";
import { setMovies } from "../../services/movie/actions";
import { clearBallot, saveBallot } from "../../services/ballot/actions";
import WelcomeMessage from "../../components/WelcomeMessage/WelcomeMessage";

const mapStateToProps = ({ ballot }) => ({
  movies: ballot.movies,
  isSavingBallot: ballot.isSaving,
  newBallot: ballot.id,
});

const mapDispatchToProps = (dispatch) => ({
  clearBallot: () => dispatch(clearBallot()),
  saveBallot: () => dispatch(saveBallot()),
});

export const NominationPage = ({
  appElement,
  clearBallot,
  movies = [],
  isSavingBallot,
  newBallot,
  saveBallot,
  history,
}) => {
  const [isConfirmationModalOpen, setConfirmationModalOpen] = useState(false);

  useEffect(() => {
    if (newBallot) {
      clearBallot();
      history.push(`ballot/${newBallot}`);
    }
  }, [newBallot]);

  return (
    <div className="content-container">
      <MovieSearch numResults={5} />
      {newBallot && (
        <div>
          Go to {window.location.href + `/ballot/${newBallot} to vote!`}
        </div>
      )}
      {movies.length > 0 ? <MovieList /> : <WelcomeMessage />}
      <div className="container--flex-row">
        <button
          className="button"
          onClick={() => {
            saveBallot();
          }}
        >
          {isSavingBallot ? "Saving..." : "Next (Voting)"}
        </button>
        <button
          className="button button--secondary"
          onClick={() => setConfirmationModalOpen(true)}
          disabled={movies.length < 1}
        >
          Start Over
        </button>
      </div>
      <VideoModal appElement={appElement} />
      <ConfirmationModal
        appElement={appElement}
        isOpen={isConfirmationModalOpen}
        onClose={() => setConfirmationModalOpen(false)}
        onConfirm={() => {
          clearBallot();
          setConfirmationModalOpen(false);
        }}
      />
    </div>
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(NominationPage);
