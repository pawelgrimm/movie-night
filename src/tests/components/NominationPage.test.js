import { expect } from "@jest/globals";
import React from "react";
import { shallow } from "enzyme";
import NominationPage from "../../components/NominationPage";

test("should render DashboardPage", () => {
  const wrapper = shallow(<NominationPage />);
  expect(wrapper).toMatchSnapshot();
});
