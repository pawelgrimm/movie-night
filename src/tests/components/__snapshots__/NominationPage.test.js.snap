// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NominationPage with movies 1`] = `
<div
  className="content-container"
>
  <MovieSearch
    numResults={5}
  />
  <Connect(NominationMovieList) />
  <div
    className="container--flex-row"
  >
    <button
      className="button"
    >
      Next (Voting)
    </button>
    <button
      className="button button--secondary"
      disabled={false}
      onClick={[Function]}
    >
      Start Over
    </button>
  </div>
  <Connect(VideoModal) />
  <ConfirmationModal
    isOpen={false}
    onClose={[Function]}
    onConfirm={[Function]}
  />
</div>
`;

exports[`should render NominationPage with no movies 1`] = `
<div
  className="content-container"
>
  <MovieSearch
    numResults={5}
  />
  <Connect(NominationMovieList) />
  <div
    className="container--flex-row"
  >
    <button
      className="button"
    >
      Next (Voting)
    </button>
    <button
      className="button button--secondary"
      disabled={true}
      onClick={[Function]}
    >
      Start Over
    </button>
  </div>
  <Connect(VideoModal) />
  <ConfirmationModal
    isOpen={false}
    onClose={[Function]}
    onConfirm={[Function]}
  />
</div>
`;
